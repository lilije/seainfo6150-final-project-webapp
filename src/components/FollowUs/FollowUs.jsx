import React from 'react';
import styles from './FollowUs.module.css';

class FollowUs extends React.Component {
  
  constructor() {
    super();
    this.submitForm = this.submitForm.bind(this);
    this.state = {
      submit: false
    }
  }
  
  displayContent = () => {
    if (this.state.submit) {
      return (
        <div className={styles.update}>
          Thanks for your subscription! You will get a notification once we update menu.
        </div>
      );
    } else {
      return (
        <div>
          <div className={styles.firstRowContentRightTop}>
            <h2>Get free recipes and special offers delivered to your inbox every week!</h2>
          </div>
          <form className={styles.firstRowContentRightCenter} onSubmit={this.submitForm}>
            <input placeholder="Email Address" className={styles.emailInput} aria-label="Email Address" maxLength="255" type="email" size="255" name="user[email]" />
            <input type="submit" name="commit" className={styles.emailSubmit} value="Submit" aria-label="Submit" data-disable-with="submit" />
          </form>
        </div>
      );
    }
  }
  
  displaySocialMedias = () => {
    const listItems = this.props.socialMedias.map((socialMedia, index) =>
      <img className={styles.socialMedia} src={socialMedia} alt="social media" key={index}/>)
    return (listItems);
  }
  
  submitForm(event) {
    event.preventDefault();
    this.setState({
      submit: true
    });
  }
  
  render() {
    return (
      <div className={styles.content}>
        <div className={styles.firstRow}>
          <div className={styles.firstRowContent}>
            <div className={styles.firstRowContentLeft}>
              <img src="https://images.unsplash.com/photo-1546173078-a7d5f578b31c?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&auto=format&fit=crop&w=934&q=80" alt="product" />
            </div>
            <div className={styles.firstRowContentRight}>
              <div>
                {this.displayContent()}
              </div>
              <div className={styles.firstRowContentRightBottom}>
                {this.displaySocialMedias()}
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default FollowUs;