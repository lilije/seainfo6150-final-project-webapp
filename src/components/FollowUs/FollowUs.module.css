.content {
  padding-bottom: 80px;
}

.firstRow {
  padding-top: 80px;
  max-width: 1200px;
  margin: 0 auto;
  padding-bottom: 160px;
}

.firstRowContent {
  display: flex;
  flex-direction: column;
}

.firstRowContentLeft {
  width: 100%;
}

img { width: 100%; }

.firstRowContentRight {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: left;
  width: 60%;
  padding-left: 80px;
}

.firstRowContentRightTop {
  text-align: center;
}

.firstRowContentRightCenter {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.emailInput {
  width: 210px;
  padding: 15px;
  border-radius: 5px;
}

.emailSubmit {
  background-color: #008600;
  color: #fff;
  border-radius: 5px;
  padding: 15px;
  cursor: pointer;
}

.firstRowContentRightBottom {
  padding-top: 30px;
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.socialMedia {
  margin-right: 20px;
  width: 50px;
  cursor: pointer;
}

.update {
  text-align: center;
  font-size: 24px;
  color: #008600;
  font-weight: 500;
}

@media screen and (min-width: 1200px) {
  .firstRowContent {
    flex-direction: row;
  }
  
  .firstRowContentLeft {
    width: 50%;
    padding: 30px;
  }
}