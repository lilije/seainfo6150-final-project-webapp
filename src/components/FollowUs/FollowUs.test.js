import React from "react";
import { render } from "@testing-library/react";
import FollowUs from "./FollowUs.jsx";

it("renders the FollowUs component correctly", () => {
  const socialMedias = ["https://www.flaticon.com/svg/static/icons/svg/1051/1051258.svg"];
  const { container } = render(<FollowUs socialMedias={socialMedias}/>);
  expect(container).toMatchSnapshot();
});