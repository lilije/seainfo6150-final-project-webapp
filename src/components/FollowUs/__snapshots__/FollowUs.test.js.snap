// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FollowUs component correctly 1`] = `
<div>
  <div
    class="content"
  >
    <div
      class="firstRow"
    >
      <div
        class="firstRowContent"
      >
        <div
          class="firstRowContentLeft"
        >
          <img
            alt="product"
            src="https://images.unsplash.com/photo-1546173078-a7d5f578b31c?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&auto=format&fit=crop&w=934&q=80"
          />
        </div>
        <div
          class="firstRowContentRight"
        >
          <div>
            <div>
              <div
                class="firstRowContentRightTop"
              >
                <h2>
                  Get free recipes and special offers delivered to your inbox every week!
                </h2>
              </div>
              <form
                class="firstRowContentRightCenter"
              >
                <input
                  aria-label="Email Address"
                  class="emailInput"
                  maxlength="255"
                  name="user[email]"
                  placeholder="Email Address"
                  size="255"
                  type="email"
                />
                <input
                  aria-label="Submit"
                  class="emailSubmit"
                  data-disable-with="submit"
                  name="commit"
                  type="submit"
                  value="Submit"
                />
              </form>
            </div>
          </div>
          <div
            class="firstRowContentRightBottom"
          >
            <img
              alt="social media"
              class="socialMedia"
              src="https://www.flaticon.com/svg/static/icons/svg/1051/1051258.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
